### DISCUSSION TITLE
How to connect a bluetooth keyboard to an esp32

### DISCUSSION THREAD

**Discussion by yimeca** [micropython/micropython#14284] [Using MicroPython]
Hi there,

I'm currently trying to make a 'mini pc' out of a [Cheap Yellow Display](https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display).

I'm currently trying to connect a bluetooth keyboard to my esp32. If I've understood correctly, aioble is for ble only, and I want to use classic bluetooth. I think that this is what is being asked [here](https://github.com/orgs/micropython/discussions/10651) and it seems that micropython doesn't support this - but I don't think that I completely understood what was being said there.

Could someone confirm this, or possibly suggest a way I can create this myself? I enjoy coding, but I'd rather not compile something.

I'm currently using [this](https://www.amazon.co.uk/Multi-Device-Bluetooth-Keyboard-Permanent-Anti-Microbial/dp/B0C2KZ8J9R/ref=sr_1_12?crid=12MA30VPWT4ZQ&dib=eyJ2IjoiMSJ9.CULmMqMwsBfLuKtME9ePzTDE6kjqFue0yS-4JZHClb9WGKmyqDU_H1K8wohkY9rlpHPuCXsyrWAY9v4xEydtkwulAK2x86pDS31MGLkSx1SPZryThUdPqpk96DbnEzTM-Mj_p5Og043AFMkdtHYyhNO2lC8NyqCXQDcJMngY9dHT4tnckN0seC1zrg2X7_jm1W741Ck0TES4pwJTDD9GUeyZp6bsf28R_4onacalAEk.6WwMGzxneDmoESaqrCt4eD1gb3qYNLJhvXwOMFccPXE&dib_tag=se&keywords=bluetooth+keyboard&qid=1711979418&sprefix=bluestoot+keyoard%2Caps%2C1691&sr=8-12) keyboard.

Thanks!

**Comment by picpic020960** [comment]
Bonjour,
Interesting project
Question about CYD : psram on board or not ?
( On the demo ,lvgl is  running but not from lv_micropython ; and lv_micropython needs psram ,no ?)
Thanks

**Comment by peterhinch** [comment]
@yimeca BLE only is supported - not Bluetooth Classic - see [the docs](https://docs.micropython.org/en/latest/library/bluetooth.html).

**Reply by yimeca** [reply]
@peterhinch I don't mind trying to create the supporting libraries myself (someone has to!). Can you recommend where I should start - or would you say it's a waste of time?

**Reply by peterhinch** [reply]
Bluetooth is entirely outside of my experience. Hopefully someone else will provide a pointer.
